File: mancala-core/src/main/java/br/com/robertomassoni/mancala/core/domain/Board.java
package br.com.robertomassoni.mancala.core.domain;

import br.com.robertomassoni.mancala.core.domain.enums.Player;
import br.com.robertomassoni.mancala.core.exception.PitNotFoundException;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.With;

import java.util.List;

@Getter
@Setter
@With
@NoArgsConstructor
@AllArgsConstructor
public class Board {
    private List<Pit> smallPits;
    private Pit bigPit;
    private Player player;

    public Pit getSmallPitById(final Integer pitId) throws Exception {
        return smallPits.stream()
                .filter(pit -> pit.getIndex().equals(pitId))
                .findFirst()
                .orElseThrow(() -> new PitNotFoundException("Small Pit not found"));
    }
}

File: mancala-web/src/app/model/Board.ts
import { Pit } from "./Pit";

export class Board {
    big_pit: Pit;
    player: string;
    small_pits: Pit[];
  
    constructor(big_pit: Pit, player: string, small_pits: Pit[]) {
      this.big_pit = big_pit;
      this.player = player;
      this.small_pits = small_pits;
    }
  }
  